src/ILLink.Tasks/CreateRuntimeRootDescriptorFile.cs

```diff
@@ -54,9 +54,43 @@ class ClassMembers
 			public HashSet<string> fields;
 		}
 
+		/// <summary>
+		/// Helper utility to track feature switch macros in header file
+		/// This type is used in a dictionary as a key
+		/// </summary>
+		readonly struct FeatureSwitchMembers
+		{
+			public string Feature { get; }
+			public string FeatureValue { get; }
+			public string FeatureDefault { get; }
+			// Unique value to track the key
+			private readonly String _key;
```

Comment:
Do we need this field? Why not implement `FeatureSwitchMembers : IEquatable<FeatureSwitchMembers>` and have
```C#
public bool Equals (FeatureSwitchMembers other) => (Feature, FeatureValue, FeatureDefault) == (other.Feature, other.FeatureValue, other.FeatureDefault);

public override bool Equals (object obj) => obj is FeatureSwitchMembers featureSwitchMembers && Equals (featureSwitchMembers);
```

Reply:
I wanted to avoid the tuple path when I realized HashCode.Combine() was only available in later .NET versions. And reverted to the simplest key implementation